src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1312,6 +1312,26 @@ public override void AddLabelForPath(string path)
             }
         }
 
+        
+        /// <summary>
+        /// Remove the labels for geometry based on their paths
+        /// </summary>
+        /// <param name="path"></param>
+        public override void ClearPathLabel(string path)
+        {
+            var nodePath = path.Contains(':') ? path.Remove(path.IndexOf(':')) : path;
+            var labelName = nodePath + TextKey;
+            lock (Model3DDictionaryMutex)
+            {
+                var sceneItemsChanged = Model3DDictionary.Remove(labelName);
+                if (sceneItemsChanged)
+                {
+                    OnSceneItemsChanged();
```

Comment:
It will only be called once as the `node` in `foreach (var node in rootWatchViewModel.Children)` is the parent `node`, and hence, `OnSceneItemsChanged()` will only be called for one node when it disconnects.